Ginger/Ginger/AnalyzerLib/AnalyzerPage.xaml.cs

```diff
@@ -84,34 +87,41 @@ public AnalyzerPage()
             SetAnalyzerItemsGridView();
 
             AnalyzerItemsGrid.DataSourceList = mIssues;
-
+            IssuesCounterLabel.Content = "Total Issues: ";
+            CriticalAndHighIssuesLabel.Content = "Total High & Critical Issues: ";
+            CanAutoFixLable.Content = "Can be Auto Fixed: ";
+            mIssues.CollectionChanged -= MIssues_CollectionChanged;
             mIssues.CollectionChanged += MIssues_CollectionChanged;
         }
 
         private void MIssues_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
         {
             Dispatcher.Invoke(() =>
-        {
-            IssuesCounterLabel.Content = "Total Issues: ";
-            IssuesCountLabel.Content = mIssues.Count();
-            if (mIssues.Any(x => x.Severity.ToString() == "High") || mIssues.Any(x => x.Severity.ToString() == "Critical"))
-            {
-                CriticalAndHighIssuesLabel.Content = "Total High & Critical Issues: ";
-                CriticalAndHighIssuesLabelCounter.Content = (mIssues.Where(x => (x.Severity.ToString() == "High")).Count() + mIssues.Where(x => (x.Severity.ToString() == "Critical")).Count());
-                CriticalAndHighIssuesLabelCounter.Foreground = new SolidColorBrush(Colors.Red);
-                CriticalAndHighIssuesLabel.Visibility = Visibility.Visible;
-            }
-            if (mIssues.Any(x => x.CanAutoFix.ToString() == "Yes"))
-            {
-                CanAutoFixLable.Content = "Can be Auto Fixed: ";
-                CanAutoFixLableCounter.Content = mIssues.Where(x => x.CanAutoFix.ToString() == "Yes").Count();
-                CanAutoFixLable.Visibility = Visibility.Visible;
-            }
-
-        });
-
+            {               
+                if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Add && e.NewItems != null)
+                {
+                    TotalIssues++;
```

Comment:
TotalIssues++ needs to be inside the the foreach loop, no?

Reply:
yes will change it.